src/internal/packager/helm/chart.go

```diff
@@ -172,44 +173,57 @@ func (h *Helm) GenerateChart(manifest types.ZarfManifest) (types.ConnectStrings,
 	spinner := message.NewProgressSpinner("Starting helm chart generation %s", manifest.Name)
 	defer spinner.Stop()
 
-	// Generate a new chart
+	// Generate a new chart.
 	tmpChart := new(chart.Chart)
 	tmpChart.Metadata = new(chart.Metadata)
 
-	// Generate a hashed chart name
+	// Generate a hashed chart name.
 	rawChartName := fmt.Sprintf("raw-%s-%s-%s", h.Cfg.Pkg.Metadata.Name, h.Component.Name, manifest.Name)
 	hasher := sha1.New()
 	hasher.Write([]byte(rawChartName))
 	tmpChart.Metadata.Name = rawChartName
 	sha1ReleaseName := hex.EncodeToString(hasher.Sum(nil))
 
-	// This is fun, increment forward in a semver-way using epoch so helm doesn't cry
+	// This is fun, increment forward in a semver-way using epoch so helm doesn't cry.
 	tmpChart.Metadata.Version = fmt.Sprintf("0.1.%d", config.GetStartTime())
 	tmpChart.Metadata.APIVersion = chart.APIVersionV1
 
-	// Add the manifest files so helm does its thing
+	var (
+		match  = []byte("{{")
+		prefix = []byte("{{`")
+		suffix = []byte("`}}")
+	)
+
+	// Add the manifest files so helm does its thing.
 	for _, file := range manifest.Files {
 		spinner.Updatef("Processing %s", file)
 		manifest := fmt.Sprintf("%s/%s", h.BasePath, file)
 		data, err := os.ReadFile(manifest)
 		if err != nil {
 			return nil, "", fmt.Errorf("unable to read manifest file %s: %w", manifest, err)
 		}
+
+		// If the template contains the match "{{", wrap it in {{` and `}} to prevent helm from trying to parse it.
+		if bytes.Contains(data, match) {
+			data = append(prefix, data...)
+			data = append(data, suffix...)
+		}
```

Comment:
This doesn't account for backticks in the templat itself.

Reply:
Is there some real-world use case that includes nested templating and backticks? I haven't seen it.

Reply:
This is a 99% solution. We cannot optimize for every single possibility, real or imagined. The released version of Zarf is broken for nested templating, and since there is an edge case for double-nested templating, this is the best way to resolve it. I have not found any example of nested or multi-nested templating + backtick escaping. It's all generally not great behavior because [kyverno](https://kyverno.io/docs/writing-policies/variables/#variables-in-helm) made a bad choice and decided to use the same templating language as the much more well-established helm.

Reply:
I think this is highly over-optimizing for an unknown edge case but added it while noting my disagreement.